dcos/subcommand.py

```diff
@@ -324,6 +385,64 @@ def _find_virtualenv(bin_directory):
     return virtualenv_path
 
 
+def _install_with_binary(
+        package_name,
+        package_dir,
+        binary_cli):
+    """
+    :param package_name: the name of the package
+    :type package_name: str
+    :param package_dir: the path to the directory in which to install the
+                          package's binary cli
+    :type package_dir: str
+    :param binary_cli: binary cli to install
+    :type binary_cli: str
+    :rtype: None
+    """
```

Comment:
We need to make sure that we handle as many failure conditions as possible. We can use `tempfile` to help us with this. I'll highlight the operations that I know can fail.